### ISSUE TITLE
Unable to bind (TCP)

### ISSUE THREAD

**Issue by paranoidjack** [jedisct1/dnscrypt-proxy#368]
Orange Pi PC with DietPi: http://www.orangepi.org/orangepibbsen/forum.php?mod=viewthread&tid=662&extra=page%3D1

```
Linux dns1 3.4.39-01-lobo #1 SMP PREEMPT Sun Oct 25 14:46:41 CET 2015 armv7l GNU/Linux

PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

Version 1.6.1 of DNScrypt fails to launch with the following error:

```
root@dns1:~# dnscrypt-proxy -R dnscrypt.eu-nl
[INFO] + DNS Security Extensions are supported
[INFO] + Provider supposedly doesn't keep logs
[NOTICE] Starting dnscrypt-proxy 1.6.1
[INFO] Generating a new session key pair
[INFO] Done
[ERROR] Unable to bind (TCP)
```

It happens no matter which interface or port is used.

Version 1.6.0 launches without any problems.

**Comment by jedisct1** [comment]
It means that something else is already listening to port 53.
Maybe a DNS resolver, such as dnsmasq.

**Comment by paranoidjack** [comment]
There's definitely nothing running that's using the same port; I've also tried binding it to different ports than 53.

```
Proto Recv-Q Send-Q Local Address           Foreign Address         State
tcp        0      0 *:ssh                   *:*                     LISTEN
tcp        0     36 10.0.0.3:ssh            10.0.0.120:51665        ESTABLISHED
tcp        0      0 10.0.0.3:ssh            10.0.0.120:51667        ESTABLISHED
tcp6       0      0 [::]:ssh                [::]:*                  LISTEN
udp        0      0 *:bootpc                *:*
udp        0      0 *:56161                 *:*
udp6       0      0 [::]:45909              [::]:*
```

**Comment by paranoidjack** [comment]
Seems the Orange Pi is at fault - 1.6.1 works without any problems on a Raspberry Pi 2

**Comment by chauffer** [comment]
Happens for me too on Banana Pi M1+, but only with certain distros. Debian Bananian has this issue, Ubuntu Armbian doesn't

**Comment by lamixer** [comment]
Please look into this. The other commenters are not in error. I had been running an older version of dnscrypt-proxy and today decided to update my libsodium and dnscrypt-proxy to the lastest, so from 1.4.0 to 1.6.1, and I ran into this error. I compiled from source and installed the prescribed way on my Ubuntu 14.04.4 virtual machine.

I think it probably could be regarded as proof that in this situation, compiling and installing 1.6.0 right over the top of 1.6.1 -- and having it WORK -- means that 1.6.1 has a problem.

```
[NOTICE] Starting dnscrypt-proxy 1.6.1
[INFO] Ephemeral keys enabled - generating a new seed
[INFO] Done
[ERROR] Unable to bind (TCP)
[NOTICE] Starting dnscrypt-proxy 1.6.0
[INFO] Ephemeral keys enabled - generating a new seed
[INFO] Done
[INFO] Server certificate #808464433 received
[INFO] This certificate looks valid
[INFO] Chosen certificate #808464433 is valid from [2015-09-11] to [2016-09-10]
[INFO] Server key fingerprint is E7AC:5C21:A4E6:6A90:B254:DD73:5229:3BA1:5BE9:8EB3:4E8F:E538:52DE:A2FB:DDB6:1357
[NOTICE] Proxying from 127.0.0.1:40 to 176.56.237.171:443
```
I changed nothing in between running 1.6.1 in a failure state and running 1.6.0 successfully. To help others reduce frustration I post this information, and look forward to 1.6.2! Thank you for creating this very useful software!

**Comment by Ardakilic** [comment]
I'm also having issue with dnscrypt, you can see at #561 with Orange Pi Zero. The formula that worked with Raspberry Pi Zero / Raspbian didn't want to work with ARMBian / OrangePi Zero. (Google brought me here by the way when I was searching why it didn't work.) Actually, @chauffer 's comment made me try other distros, such as DietPi and Armbian Ubuntu. DietPi had the same issue but ARMBian Ubuntu worked as it should. So the source of the issue may be the kernel + Debian base itself. I'd suggest you guys try the same steps with Ubuntu.

**Comment by Ardakilic** [comment]
Catch-up , please see the discussion #561 , the newly-released 1.9.3 version (and hopefully possible future versions) fixed the problem for me on OPi Zero ARMBian.
